assets/js/translator.js

```diff
@@ -1111,14 +1114,71 @@ function setRecentDstLangs(langs /*: string[] */) {
     return langs;
 }
 
+function dictionaryLookup() {
+    if(modeEnabled('lookup')) {
+        if(currentLookupRequest) {
+            currentLookupRequest.abort();
+        }
+
+        $('#dictionaryLookupResult').empty();
+        var wordToLookup = $('#originalText').val().trim().split(' '); // eslint-disable-line newline-per-chained-call
+        if(wordToLookup.length <= 3 && wordToLookup[0] !== '') {
+            currentLookupRequest = callApy({
+                data: {
+                    'langpair': curSrcLang + '|' + curDstLang,
+                    'q': $('#originalText').val()
+                },
+                success: handleDictionaryLookupSuccessResponse,
+                error: handleDictionaryLookupErrorResponse,
+                complete: function () {
+                    ajaxComplete();
+                    currentLookupRequest = undefined;
+                }
+            }, '/dictionaryLookup');
+        }
+    }
+
+    function handleDictionaryLookupSuccessResponse(data) {
+        $('#dictionaryLookup').removeClass('hide');
+        if($('#currentWordForLookup').length === 0) {
+            var currentWord = $('<span id="currentWordForLookup">');
+            currentWord.html($('#originalText').val());
+            $('#dictTranslation').html($('#dictTranslation').html().replace('{{word}}', ''));
+            $('#dictTranslation').append(currentWord);
+        }
+        else {
+            $('#currentWordForLookup').html($('#originalText').val());
+        }
+        for(var analysisType in data) {
+            var values = $('<ul id="dictionaryTranslationContents">');
+            for(var translationContent = 0; translationContent < data[analysisType].length; translationContent++) {
+                var livalues = $('<li>');
+                livalues.html(data[analysisType][translationContent].replace('#', ''));
+                values.append(livalues);
+            }
+            var dlSpan = $('<span id="dictDLHeaders">');
+            dlSpan.attr({'data-text': 'POS_' + analysisType});
+            dlSpan.html(dynamicLocalizations['part_of_speech'][analysisType]);
```

Comment:
Do we also want to provide additional information like the gender of nouns or the paradigm for verb conjugation? In my opinion this would be very helpful; the former is already done by other services, the latter is something I always wish they did. (Would be a change to APy too.)